### ISSUE TITLE
Snapshot Size

### ISSUE THREAD

**Issue by thwaller** [bit-team/backintime#991]
Enhancement Idea:
I was looking for an option to show the size of the snapshots, but could not. Is this something you may consider adding?

What I am looking to see is the scale of each snapshot. For example, Seeing a large snapshot shows me it is largely inclusive whereas a small size snapshot is more supplemental. Basically, a visual to full backups vs incremental or differential backups. I would find this useful in managing backup history kept.

**Comment by colinl** [comment]
I too would find it very convenient to see the additional disc space used after each backup if that were possible.  Approximate would be good enough.

**Comment by Germar** [comment]
I totally agree this would be a nice to have feature. But technically this is not possible. BiT creates snapshots with hard-links. So if you backup a file that didn't change BiT will just create an other link to the already stored data. If you remove the link, the data is still available. Only when you remove the last link to the data the space is considered to be available by the filesystem again.
So you can't count which snapshot uses the data and also you can't tell which snapshot was the first to use the data. And when you delete the first snapshot, you would have to know which snapshot the data is still attached to.
The only way to know the exact size of each snapshot would be to run something like `ncdu` which scans the snapshots in chronology order. I wasn't able to force `ncdu` to do so last time I checked. Also tools like `ncdu` take ages on huge backups with lots of snapshots...

**Comment by thwaller** [comment]
It appears my understanding of the process in somewhat incomplete. Rather then getting into a discussion here, is there a document you have that outlines more specifically the methodology of the backup process? I would rather start fresh as my mind is used to the norm of a full backup followed by change based backups tied to the last full backup.

**Comment by Germar** [comment]
This [FAQ](https://github.com/bit-team/backintime/wiki/FAQ#how-does-snapshots-with-hard-links-work) does shed a light on how snapshots with hard-links work

**Comment by thwaller** [comment]
Is it possible to mark which snapshots are full backups?

**Comment by Germar** [comment]
There is no 'full backup'. BiT only sync those files which are not already in the previous snapshot. All others will be hard-linked. There is no way of knowing if no files at all where hard-linked as this is done by `rsync`

**Comment by DonEdwards** [comment]
"Is it possible to mark which snapshots are full backups?"

All of them are.

**Comment by thwaller** [comment]
> "Is it possible to mark which snapshots are full backups?"
> 
> All of them are.

Let me rephrase ... is there a way to mark those with no hardlinks. They are not all full backups, they are all complete backups though. Germar addressed the question making it no longer valid given the context, but it is quite misleading to call all of the backups full backups.

**Comment by Codeberg-AsGithubAlternative-buhtz** [comment]
@emtiu
I was referenced to this Issue by https://github.com/bit-team/backintime/pull/1162

I **vote for close** because the feature is out of scope of BiT.

If this wish comes up in the future again I would do some investigations and then create an entry in the Wiki- or the documentation about how to solve that very easy without using Bit.

**Comment by emtiu** [comment]
Still a useful feature idea – closing as nearly-duplicate of #406.
